src/instrumenter/annotations.ts

```diff
@@ -267,242 +319,253 @@ type RawMetaData = {
+                target
+            );
+        }
+    } else {
+        if (
+            annotation.type !== AnnotationType.IfUpdated &&
+            annotation.type !== AnnotationType.IfAssigned &&
+            /**
+             * @todo Figure out handling the difference between syntax sugar macros for state vars
+             * and contract level macros. This is not yet clear.
+             */
+            annotation.type !== AnnotationType.Macro
+        ) {
+            throw new UnsupportedByTargetError(
+                `The "${annotation.type}" annotation is not applicable to state variables`,
+                annotation.original,
+                annotation.annotationFileRange,
+                target
+            );
         }
-    }
 
-    private findAnnotations(
-        raw: StructuredDocumentation,
-        target: AnnotationTarget,
-        source: string,
-        filters: AnnotationFilterOptions
-    ): AnnotationMetaData[] {
-        const rxType = filters.type === undefined ? undefined : new RegExp(filters.type);
-        const rxMsg = filters.message === undefined ? undefined : new RegExp(filters.message);
+        if (!(target.vScope instanceof ContractDefinition)) {
+            throw new UnsupportedByTargetError(
+                `The "${annotation.type}" annotation is only applicable to state variables`,
+                annotation.original,
+                annotation.annotationFileRange,
+                target
+            );
+        }
+    }
+}
 
-        const sourceInfo = raw.sourceInfo;
+function findAnnotations(
+    raw: StructuredDocumentation,
+    target: AnnotationTarget,
+    source: string,
+    ctx: AnnotationExtractionContext
+): AnnotationMetaData[] {
+    const rxType =
+        ctx.filterOptions.type === undefined ? undefined : new RegExp(ctx.filterOptions.type);
 
-        const meta: RawMetaData = {
-            target: target,
-            node: raw,
-            text: raw.extractSourceFragment(source),
-            loc: [sourceInfo.offset, sourceInfo.length]
-        };
+    const rxMsg =
+        ctx.filterOptions.message === undefined ? undefined : new RegExp(ctx.filterOptions.message);
 
-        const result: AnnotationMetaData[] = [];
+    const sourceInfo = raw.sourceInfo;
 
-        const rx =
-            /\s*(\*|\/\/\/)\s*#?(if_succeeds|if_updated|if_assigned|invariant|assert|try|require|define\s*[a-zA-Z0-9_]*\s*\([^)]*\))/g;
+    const meta: RawMetaData = {
+        target: target,
+        node: raw,
+        text: raw.extractSourceFragment(source),
+        loc: [sourceInfo.offset, sourceInfo.length]
+    };
 
-        let match = rx.exec(meta.text);
+    const result: AnnotationMetaData[] = [];
 
-        while (match !== null) {
-            const annotation = this.makeAnnotationFromMatch(match, meta, source);
+    const rx =
+        /\s*(\*|\/\/\/)\s*#?(if_succeeds|if_updated|if_assigned|invariant|assert|try|require|define|macro\s*[a-zA-Z0-9_]*\s*\([^)]*\))/g;
```

Comment:
Nit: The modification of the pattern lost the requirement to have a `word(...)` after `define`

Reply:
Addressed in https://github.com/ConsenSys/scribble/pull/96/commits/24012c080b685c49cbcc930f86a107cfee582a84